ledger/src/blockstore.rs

```diff
@@ -1192,7 +1182,7 @@ impl Blockstore {
                 return false;
             }
 
-            if !Blockstore::should_insert_coding_shred(&shred, &self.last_root) {
+            if !Blockstore::should_insert_coding_shred(&shred, self.max_root()) {
```

Comment:
The former `RwLock` / now `AtomicU64` is getting re-called for every single shred that is inserted. I'm thinking we could fetch this number once at the start of the loop that iterates through every shred in `Blockstore::do_insert_shreds()`. Looking at metrics for my node for the last 24 hours, I'm seeing something like 4k shreds per second (`blockstore-insert-shreds.num_shreds` / 2 since this metric is reported every 2s).

However, glancing at `recv-window-insert-shreds.run_insert_count`, I see a number on the order of 1000's which would indicate that on average, we're calling `Blockstore::insert_shreds()` with a small number of shreds. So, the gains of reading once may not be so big (factor of 2), but wondering if having so many calls with so few shreds per call is intended / optimal behavior